src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -69,19 +79,52 @@ public static string CreateReferenceId(ISymbol symbol)
         /// <summary>
         /// Gets all declaration symbols that match the declaration id string
         /// </summary>
-        public static IEnumerable<ISymbol> GetSymbolsForDeclarationId(string id, Compilation compilation)
+        public static ImmutableArray<ISymbol> GetSymbolsForDeclarationId(string id, Compilation compilation)
         {
-            List<ISymbol> results;
-            TryGetSymbolsForDeclarationId(id, compilation, out results);
-            return results;
+            if (id == null)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (compilation == null)
+            {
+                throw new ArgumentNullException(nameof(compilation));
+            }
+
+            var results = s_symbolListPool.Allocate();
+            try
+            {
+                Parser.ParseDeclaredSymbolId(id, compilation, results);
+                return results.ToImmutableArray();
+            }
+            finally
+            {
+                s_symbolListPool.ClearAndFree(results);
+            }
         }
 
         /// <summary>
-        /// Try get all declaration symbols that match the declaration id string
+        /// Try to get all the declaration symbols that match the declaration id string.
+        /// Returns true if atleast one symbol matches.
         /// </summary>
-        public static bool TryGetSymbolsForDeclarationId(string id, Compilation compilation, out List<ISymbol> results)
+        /// <remarks>This API is available for possible of reuse of results lists.</remarks>
+        public static bool TryGetSymbolsForDeclarationId(string id, Compilation compilation, List<ISymbol> results)
```

Comment:
:bulb: I'm not sure I'm ready for this method to be in the public API, so I would consider making it `internal`.

I believe `GetSymbolsForDeclarationId` covers the essential external use cases already.